source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -143,9 +151,31 @@ public Task<TResource> Refresh(TResource resource)
                 if (resource == null) throw new ArgumentNullException("resource");
                 return Get(resource.Id);
             }
-        }
 
-        // ReSharper restore MemberCanBePrivate.Local
+            protected Dictionary<string, object> CreateSpacesParameters(bool includeGlobal, params string[] spaces)
+            {
+                ValidateSpacesParameters(spaces);
+
+                return new Dictionary<string, object>
+                {
+                    ["includeGlobal"] = includeGlobal,
+                    ["spaces"] = spaces
+                };
+            }
+
+            void ValidateSpacesParameters(params string[] spaces)
+            {
+                if (LimitedToSpacesParameters == null) return;
+                var previouslyDefinedSpaceIds = LimitedToSpacesParameters["spaces"] as string[];
+                var previouslyDefinedSpaceIdsSet = new HashSet<string>(previouslyDefinedSpaceIds);
```

Comment:
Don't cast here... just store a more specific type. Just create your own type, and map to the parameter dictionary at request time

Reply:
Ah I see why you did it this way - the basic repository is used by non-space scoped repositories as well. Maybe this indicates we need another base class in that type hierarchy?